lang/cs/Org.Apache.REEF.Tests/Functional/Failure/EvaluatorWithRunningTaskImmediatePoisonTest.cs

```diff
@@ -0,0 +1,138 @@
+
+        private const string FailedEvaluatorMessage = "I have succeeded in seeing a failed evaluator.";
+
+        [Fact]
+        [Trait("Description", "Test evaluator failure by injecting immediate fault in task start handler.")]
+        public void TestPoisonedTaskStartHandlerImmediate()
+        {
+            var testFolder = DefaultRuntimeFolder + TestId;
+            TestRun(DriverConfigurations(), typeof(PoisonedEvaluatorDriver), 1, "poisonedTaskStartImmediateTest", "local", testFolder);
+            ValidateMessageSuccessfullyLoggedForDriver(FailedEvaluatorMessage, testFolder);
+
+            // verify that no unexpected events happened
+            ValidateMessageSuccessfullyLoggedForDriver(BasePoisonedEvaluatorDriver.UnexpectedClosedContext, testFolder, 0);
+            ValidateMessageSuccessfullyLoggedForDriver(BasePoisonedEvaluatorDriver.UnexpectedFailedContext, testFolder, 0);
+            ValidateMessageSuccessfullyLoggedForDriver(BasePoisonedEvaluatorDriver.UnexpectedCompletedTask, testFolder, 0);
+            ValidateMessageSuccessfullyLoggedForDriver(BasePoisonedEvaluatorDriver.UnexpectedFailedTask, testFolder, 0);
+            CleanUp(testFolder);
+        }
+
+        private static IConfiguration DriverConfigurations()
+        {
+            return DriverConfiguration.ConfigurationModule
+                .Set(DriverConfiguration.OnDriverStarted, GenericType<PoisonedEvaluatorDriver>.Class)
+                .Set(DriverConfiguration.OnEvaluatorAllocated, GenericType<PoisonedEvaluatorDriver>.Class)
+                .Set(DriverConfiguration.OnEvaluatorFailed, GenericType<PoisonedEvaluatorDriver>.Class)
+                .Set(DriverConfiguration.OnContextActive, GenericType<PoisonedEvaluatorDriver>.Class)
+                .Set(DriverConfiguration.OnContextClosed, GenericType<PoisonedEvaluatorDriver>.Class)
+                .Set(DriverConfiguration.OnContextFailed, GenericType<PoisonedEvaluatorDriver>.Class)
+                .Set(DriverConfiguration.OnTaskCompleted, GenericType<PoisonedEvaluatorDriver>.Class)
+                .Set(DriverConfiguration.OnTaskFailed, GenericType<PoisonedEvaluatorDriver>.Class)
+                .Build();
+        }
+
+        private sealed class PoisonedEvaluatorDriver :
+            BasePoisonedEvaluatorDriver,
+            IObserver<IFailedEvaluator>,
+            IObserver<IActiveContext>
+        {
+            [Inject]
+            private PoisonedEvaluatorDriver(IEvaluatorRequestor requestor) : base(requestor)
+            {
+            }
+
+            public void OnNext(IActiveContext value)
+            {
+                var taskConfig = TaskConfiguration.ConfigurationModule
+                    .Set(TaskConfiguration.Identifier, TaskId)
+                    .Set(TaskConfiguration.Task, GenericType<SleepTask>.Class)
+                    .Set(TaskConfiguration.OnClose, GenericType<SleepTask>.Class)
```

Comment:
If the Driver issues the shutdown, then yes, the Task will get a close event. If the Task ends on its own, there shouldn't be an event raised to it.